model-server/src/main/kotlin/org/modelix/model/server/handlers/IRepositoriesManager.kt

```diff
@@ -55,6 +57,17 @@ interface IRepositoriesManager {
      */
     fun mergeChangesBlocking(branch: BranchReference, newVersionHash: String): String
     suspend fun computeDelta(versionHash: String, baseVersionHash: String?): ObjectData
+
+    suspend fun <R> runWithRepository(repository: RepositoryId, body: suspend () -> R): R
+    fun findObject(hash: String): List<ObjectInRepository>
+
+    /**
+     * The data of a repository is stored separately from other repositories, but that wasn't always the case.
+     * For backwards compatibility, existing repositories remain in the global storage.
+     *
+     * @return null if the repository doesn't exist
```

Comment:
I'm not sure if I like this use of nullability. Other opinions on that?

Reply:
Without an alternative there is nothing to discuss.

Reply:
An exception would be the alternative

Reply:
That would require an additional check, if a repository exists before calling this method. That's not a better alternative. I don't understand why a null value is an issue at all.

Reply:
Not an issue. Just a question of what would be more intuitive for users of this interface.